### DISCUSSION TITLE
StandupsListTest crashes

### DISCUSSION THREAD

**Discussion by easiwriter** [pointfreeco/swift-composable-architecture#3339] [General]
### Description

Running StandupsListTest crashes consistently at line 521 of the extension to OrderedSet. It looks to me as though some code is missing from the guard statement because when _elements has zero items it attempts to remove item[0].

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

I'd expect the test to complete normally

### Actual behavior

It crashes as described above

### Steps to reproduce

Just run the test

### The Composable Architecture version information

1.14.0

### Destination operating system

IOS 17

### Xcode version information

Xcode 15.5

### Swift Compiler version information

```shell
SWift 5
```

**Comment by mbrandonw** [comment]
Hi @easiwriter, I am not able to reproduce this. It passes just fine for me:

<img width="1415" alt="image" src="https://github.com/user-attachments/assets/f4bdc9e0-72f6-433c-b1cf-9d6369803c14">

And fwiw, these tests run on CI with every PR and they have been passing for a long time. Can you provide some proof of the crash?

I'm not sure this is an issue with the library, and so for now I am going to convert it to a discussion.  Feel free to continue the conversation over there.

**Reply by easiwriter** [reply]
Hi

Thanks for the reply. The screenshot you sent is for the SyncUps app, not Standups. As I mention in another discussion the SyncUps project won't load for me - Xcode just hangs. I have tried all the usual ways to get it to work - clean build, deleting derived data, reinstalling the package. 

I will provide proof of the Standups crash tomorrow - I'm on GMT.

Cheers
Keith (Easiwriter)

**Comment by easiwriter** [comment]
Here's the code from the list test:

`@MainActor
final class StandupsListTests: XCTestCase {
    
    func testAddStandup() async {
        let store = TestStore(initialState: StandupsListFeature.State(), reducer: {
            StandupsListFeature()
        }, withDependencies: {
            $0.uuid = .incrementing
        })
        
        var standup: Standup = Standup(
            id: UUID(0),
            attendees: [.init(id: UUID(1))]
        )
        await store.send(.addButtonTapped) {
            $0.addStandup = StandupFormFeature.State(
                standup: standup
            )
        }
        
        standup.title = "Lisandro's Team Morning Sync"
        await store.send(.addStandup(.presented(.set(\.$standup, standup)))) {
            $0.addStandup?.standup.title = "Lisandro's Team Morning Sync"
        }
        
        await store.send(.saveStandupButtonTapped) {
            $0.standups[0] = Standup(
                id: UUID(0),
                attendees: [.init(id: UUID(1))],
                title: "Lisandro's Team Morning Sync"
            )
            $0.addStandup = nil
        }
    }
`
Running the test fails here:

<img width="1209" alt="Screenshot 2024-09-01 at 08 40 40" src="https://github.com/user-attachments/assets/dcc373c9-6219-4bfc-b999-dcdee7731af3">

The crash occurs here:

<img width="1159" alt="Screenshot 2024-09-01 at 08 39 16" src="https://github.com/user-attachments/assets/10a9a180-3fd9-4ec4-a63f-677dcd721a10">

Here's the stack:

<img width="454" alt="Screenshot 2024-09-01 at 08 39 58" src="https://github.com/user-attachments/assets/eaa18d65-e87d-415c-9956-fdc07fbc9585">

I first hit the problem when I tried testing my own code that is modelled on your code. Here's mine:

<img width="1124" alt="Screenshot 2024-09-01 at 08 48 51" src="https://github.com/user-attachments/assets/882890b9-0e2f-4251-b088-1994fb633de4">

This fails in exactly the same way.

Regards
Keith

**Reply by mbrandonw** [reply]
Hi @easiwriter, I'm sorry but I'm not really understanding your situation. There is no "Standups" project in this repo. There is only "SyncUps". And some of the code you are sharing here is not in our repo, such as this `testProjectButtonTapped` method you are showing in your last screenshot. It also seems like you are possibly using SwiftData in this code, and that is not something we have in our repo.

So, as far as I can tell, this is an issue with your code, not an issue with the code in the library. And if you are having an issue with your code, then you should describe it as such instead of saying that there is a problem with our code. And it would be best if you actually provide a project that demonstrated the problem, instead of sharing screenshots.

**Reply by easiwriter** [reply]
Hi Brandon,
The Standups app is the vehicle you use in your video series here: https://www.pointfree.co/collections/tours/composable-architecture-1-0. I realise the app is not in the repo, so I went looking for at and found it here :https://github.com/lisandro/standups-composable-arch. The crash takes place in SwiftCollections in the OrderedSet code. The code is the fails is in TestAddStandup and it is this that I replicate in my code. It is not using SwiftData, though will be in the near future. 
So it is my belief that there is a problem in SwiftCollections as shown in the screenshots.
Best
Keith

**Reply by mbrandonw** [reply]
Hi @easiwriter, that repo is not ours. All of our episode sample code can be found here: https://github.com/pointfreeco/episode-code-samples/

While we try to make episodes as complete as possible, mistakes do creep in every once in a while, and because the episodes are frozen in time it's not possible to update them after the fact. If you want to see the correct and final version of the code, I recommend looking at the version of SyncUps in the repo.

If you want to get the test passing in the code base you are working in, you can update the assertion to be:

```swift
$0.standups = [
  Standup(
    id: UUID(0),
    attendees: [Attendee(id: UUID(1))],
    title: "Point-Free Morning Sync"
  )
]
```

**Reply by easiwriter** [reply]
Hi Brandon

Thanks for this, and for the alternative code. The thing that bothers me is that the original code highlights an error in SwiftCollections. This was not the case in the video where the test worked, but is the case now with v1.14 of the library. The screenshot I sent shows this if you look at the stack values in the call. It looks as though some code has gone missing from the guard. 

Regards
Keith

**Reply by stephencelis** [reply]
@easiwriter The issue is not in SwiftCollections, but it was a bug in our IdentifiedArray type that uses SwiftCollections under the hood. This bug has been fixed since.
